test-utils/src/approx_eq.rs

```diff
@@ -64,12 +64,12 @@ macro_rules! assert_approx_eq {
     ($t:ty, $left:expr, $right:expr, epsilon = $epsilon:expr, ulps = $ulps:expr $(,)?) => {{
         let epsilon = $epsilon;
         let ulps = $ulps;
-        let left = $left;
-        let right = $right;
+        let left = &$left;
+        let right = &$right;
         let mut left_iter =
-            $crate::ApproxEqIter::<$t>::indexed_iter_logical_order(&left, Vec::new());
+            $crate::ApproxEqIter::<$t>::indexed_iter_logical_order(left, Vec::new());
```

Comment:
it is necessary, without it the macro takes the arguments by value instead of by reference (because of the prior assignment some lines above), which may require to clone them before calling the macro. the change aligns the `assert_approx_eq!` macro with the behavior of other macros like `assert_eq!`.